File: dropwizard_2.9.1/src/main/scala/com/yammer/dropwizard/config/RequestLogHandlerFactory.scala
package com.yammer.dropwizard.config

import com.codahale.fig.Configuration
import org.eclipse.jetty.server.handler.RequestLogHandler
import com.yammer.dropwizard.jetty.AsyncRequestLog

object RequestLogHandlerFactory {
  def buildHandler(implicit config: Configuration) = {
    val log = new AsyncRequestLog(
      config("request_log.filename").asRequired[String],
      config("request_log.retain_days").asRequired[Int]
    )
    
    val handler = new RequestLogHandler()
    handler.setRequestLog(log)
    handler
  }
}

File: dropwizard/src/main/java/com/yammer/dropwizard/config/RequestLogHandlerFactory.java
package com.yammer.dropwizard.config;

import com.yammer.dropwizard.jetty.AsyncRequestLog;
import org.eclipse.jetty.server.handler.RequestLogHandler;

import static com.yammer.dropwizard.config.HttpConfiguration.RequestLogConfiguration;

// TODO: 11/7/11 <coda> -- document RequestLogHandlerFactory
// TODO: 11/7/11 <coda> -- test RequestLogHandlerFactory

public class RequestLogHandlerFactory {
    private final RequestLogConfiguration config;

    public RequestLogHandlerFactory(RequestLogConfiguration config) {
        this.config = config;
    }
    
    public boolean isEnabled() {
        return config.isEnabled();
    }

    public RequestLogHandler build() {
        final RequestLogHandler handler = new RequestLogHandler();
        handler.setRequestLog(new AsyncRequestLog(config.getFilenamePattern(),
                                                  config.getRetainedFileCount()));
        return handler;
    }
}
